src/gui/StyleSpec.h

```diff
@@ -106,6 +117,29 @@ class StyleSpec
 		return color;
 	}
 
+	video::ITexture *getTexture(Property prop, ISimpleTextureSource *tsrc,
+			video::ITexture *def) const
+	{
+		const auto &val = properties[prop];
+		if (val.empty()) {
+			return def;
+		}
+
+		video::ITexture *texture = tsrc->getTexture(val);
```

Comment:
Shouldn't this use a texture source? `ITextureSource`, I think it's called

Test with texture modifiers